### DISCUSSION TITLE
Repeat Feature on RowsDefinitions (Grid)

### DISCUSSION THREAD

**Discussion by chandra-arifin** [CommunityToolkit/Maui.Markup#99] [New Feature Discussions]
```
`Grid grid = new Grid
        {
            Padding = new(20),
            ColumnSpacing = 20,
            RowSpacing = 20,

            RowDefinitions = Rows.Define(Auto, Auto, Auto, Auto, Auto, Auto, Auto, Auto, Auto, 35, Auto),
            ColumnDefinitions = Columns.Define(Stars(.30), Stars(.70)),
        };
```

maybe the RowDefinitions can have feature like Css Grid:

```
display: grid;
grid-template-columns: repeat(2, 50px 1fr) 100px;
grid-gap: 5px;
box-sizing: border-box;
height: 200px;...

```

at the css grid, there is "repeat" function, can this library have this feature too ? maybe? ;p

just asking.
maybe like:

`RowDefinitions = Rows.Define(Repeat(Auto, 9), 35, Auto),`

**Comment by mrlacey** [comment]
A `Repeat` function might simplify the creation of duplicate definitions, but it's unclear what each of those rows represents and how they're referenced.

I'm not sure this helps in the future maintainability of the code.

If I want to insert a new 4th row with a fixed height, changing from `RowDefinitions = Rows.Define(Repeat(Auto, 9), 35, Auto),` to `RowDefinitions = Rows.Define(Repeat(Auto, 3), 100, Repeat(Auto, 6), 35, Auto),` doesn't help with needing to update all the other places where the Row is set.

Given the example of using an enum to create named rows, (such as below,) is there any way to combine the use of named row definitions and wanting to simplify giving them all (or some of them) the  same value?

```cs
	enum Row { Title, Description, BottomPadding }

		RowDefinitions = Rows.Define(
			(Row.Title, 20),
			(Row.Description, 20),
			(Row.BottomPadding, 1)),
```


Has this been considered by anyone?
Or is it even something anyone has encountered or been concerned about?

**Reply by bijington** [reply]
Sorry I thought I had responded in a similar way to this but clearly I either dreamt it, didn't hit the Reply button or something else. I agree that I dislike that lack of identification of rows/columns and how they fit into the UI.

**Comment by CliffAgius** [comment]
Closed and answered
